sickbeard/search.py

```diff
@@ -565,7 +565,10 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
 
         # Update the cache if a manual search is being runned
         if manual_snatch:
-            results = curProvider.cache.update_cache_manual_search(searchResults[curEp])
+            results = False
+            for searched_episode in episodes:
+                results = curProvider.cache.update_cache_manual_search(searchResults[searched_episode.episode])
+
             if results:
                 # If we have at least a result from one provider, it's good enough to be marked as result
                 finalResults.append(results)
```

Comment:
As discussed we're collecting a list of bools here? With what purpose? In line 732 is picked up again with finalResults += set(multiResults.values()). So then results are added to it?

I'll look further into this, if we can agree the solution for the caching is the correct one. I haven't really looked into the finalResults, so will do that first.